pkg/distributor/distributor.go

```diff
@@ -666,7 +662,8 @@ func (d *Distributor) validateSeries(nowt time.Time, ts mimirpb.PreallocTimeseri
 	return nil
 }
 
-func (d *Distributor) wrapPushWithMiddlewares(next push.Func) push.Func {
+// wrapPushWithMiddlewares returns push function wrapped in all Distributor's middlewares.
+func (d *Distributor) wrapPushWithMiddlewares(distributorWrapper func(next push.Func) push.Func, next push.Func) push.Func {
```

Comment:
non-blocking nit: I think the name `distributorWrapper` is a but confusing because it's just another middleware like all the others, I'd name it something like `externalMiddleware`

Reply:
Renamed.